### ISSUE TITLE
Add additional lecture to JS and A11y module

### ISSUE THREAD

**Issue by huyenltnguyen** [freeCodeCamp/CurriculumExpansion#966]
## Description

Related to https://github.com/freeCodeCamp/CurriculumExpansion/issues/944.

TL;DR:
- Review the `What Is the Purpose of WAI-ARIA, and How Does It Work?` lecture and remove any concepts that will be covered later in the JS A11y module.
- Assess the removed content to determine if it should be incorporated into the JS-A11y lectures, especially the one that covers tabs (``What Is the `aria-controls` Attribute, and How Does It Work?``) since we have a `workshop-planets-tablist` coming right after.
- Add a new intro lecture to the JS A11y module.

---

In the HTML-A11y module, we have the `Introduction to ARIA attributes` block, within which, we have the `What Is the Purpose of WAI-ARIA, and How Does It Work?` lecture that introduces the following new roles and ARIA attributes:
- role=tablist
- role=tab
- role=tabpanel
- aria-label
- aria-labelledby
- aria-controls
- aria-selected
- aria-expanded

Based on the HTML A11y review page, out of those attributes only `aria-label` and `aria-labelledby` are covered in the module. I think we should consider trimming the lecture to focus on these, and move the other attributes to the JS-A11y module. This will make the introduction less overwhelming and keep the concepts fresh when campers revisit them in more depth later.

Link to the lecture file: https://github.com/freeCodeCamp/freeCodeCamp/blob/main/curriculum/challenges/english/25-front-end-development/lecture-introduction-to-aria/672a53cf67140d8cd85d4b0f.md.

Link to the HTML A11y structure for an overview of the module: https://github.com/freeCodeCamp/freeCodeCamp/blob/main/curriculum/superblock-structure/full-stack.json#L127-L140.

---

In the JS-A11y module, we start off with `What Is the aria-expanded Attribute, and How Does It Work?`. This feels a bit abrupt, since campers have been working almost exclusively with JavaScript in the previous blocks and modules, and are suddenly met with an HTML-focused lesson.

The last lecture of the lecture block (``What Is the `aria-controls` Attribute, and How Does It Work?``) touches the following attributes:
- role=tablist
- role=tab
- role=tabpanel
- aria-selected
- aria-controls
- hidden

Though the roles aren't explained in much detailed (I guess this is because we previously had the intro lecture in the same block, so there is no need to re-explain things).

Additionally, I think we probably should add a short lecture before `What Is the aria-expanded Attribute, and How Does It Work?`, explaining why campers need to learn this section. 

Perhaps the lecture would be `How Do You Make Dynamic and Interactive Content Accessible?`, and we start with:

> You previously learned how semantic markup and ARIA attributes make content accessible. In real-world apps, content isn’t static, and pages change dynamically through JavaScript needs to be reflected on the HTML so that screen readers can accurately convey the updated state to users. Without this, people relying on these tools might never know that something changed, or be given outdated or misleading information.
>
> // Examples of some aria states like `aria-expanded`, `aria-checked`, or `aria-selected`.
> // The goal here is to show how these attribute values should be updated in response to user interactions (such as a click event). We keep it simple here, as the upcoming lectures will cover the details in depth.

Link to the ``What Is the `aria-controls` Attribute, and How Does It Work?`` lecture file: https://github.com/freeCodeCamp/freeCodeCamp/blob/main/curriculum/challenges/english/25-front-end-development/lecture-understanding-aria-expanded-aria-live-and-common-aria-states/672a5507d857a891139abc7f.md

Link to the JS A11y structure for an overview of the module: https://github.com/freeCodeCamp/freeCodeCamp/blob/main/curriculum/superblock-structure/full-stack.json#L861-L882.

---

To summarize, my proposal is:
- [ ] Review the `What Is the Purpose of WAI-ARIA, and How Does It Work?` lecture and remove any concepts that will be covered later. - https://github.com/freeCodeCamp/freeCodeCamp/issues/61781
- [ ] Assess the removed content to determine if it should be incorporated into the JS-A11y lectures, especially the one that covers tabs (``What Is the `aria-controls` Attribute, and How Does It Work?``) since we have a `workshop-planets-tablist` coming right after. - https://github.com/freeCodeCamp/freeCodeCamp/pull/61782
- [ ] Add a new intro lecture to the JS A11y module. - https://github.com/freeCodeCamp/freeCodeCamp/pull/61817

**Comment by jdwilkin4** [comment]
Here are my thoughts 👍🏾 

> Review the What Is the Purpose of WAI-ARIA, and How Does It Work? lecture and remove any concepts that will be covered later in the JS A11y module.

By looking at the lecture [here](https://www.freecodecamp.org/learn/full-stack-developer/lecture-introduction-to-aria/what-is-the-purpose-of-wai-aria), we can cut out the last part about tab panels. Then the lecture can just end with the following closing:

> To get the best out of WAI-ARIA, try to stick with native HTML whenever possible since it generally provides more accessibility out of the box.

> Use WAI-ARIA only when HTML falls short, and don’t forget to test with assistive technologies like screen readers, or have people with disabilities test your work. Also, make sure your WAI-ARIA states and properties update with the content in real time. Avoid overusing ARIA as it can be confusing many times.

That change can probably be opened up for contribution. 

> Though the roles aren't explained in much detailed (I guess this is because we previously had the intro lecture in the same block, so there is no need to re-explain things).

Yeah, we can provide more context in that lecture


> Additionally, I think we probably should add a short lecture before What Is the aria-expanded Attribute, and How Does It Work?, explaining why campers need to learn this section.

> Perhaps the lecture would be How Do You Make Dynamic and Interactive Content Accessible?, and we start with:

Yeah, that should work

**Comment by huyenltnguyen** [comment]
I'm going to bump this to High Priority and hold #944 to get this one in.

I think it's better to have an explanation so campers won't be surprise when they start this module and land on HTML lectures.
